src/Schema/Comparator.php

```diff
@@ -21,23 +21,29 @@
 class Comparator
 {
     /**
+     * Returns a SchemaDiff object containing the differences between the schemas $fromSchema and $toSchema.
+     *
+     * The returned differences are returned in such a way that they contain the
+     * operations to change the schema stored in $fromSchema to the schema that is
+     * stored in $toSchema.
+     *
+     * This method is marked as static to avoid a BC break but should be called
+     * non-statically since it will not be static in the next major release.
+     *
      * @return SchemaDiff
      *
      * @throws SchemaException
      */
     public static function compareSchemas(Schema $fromSchema, Schema $toSchema)
     {
-        $c = new self();
-
-        return $c->compare($fromSchema, $toSchema);
+        /** @psalm-suppress DeprecatedMethod */
+        return (new self())->compare($fromSchema, $toSchema);
     }
 
     /**
-     * Returns a SchemaDiff object containing the differences between the schemas $fromSchema and $toSchema.
+     * This method is deprecated and will be removed in the next major release.
      *
-     * The returned differences are returned in such a way that they contain the
-     * operations to change the schema stored in $fromSchema to the schema that is
-     * stored in $toSchema.
+     * @deprecated Use non-static call to {@link compareSchemas()} instead.
```

Comment:
Let's make sure the deprecated method doesn't contain any logic except for calling the recommended method.